lib: lte_link_control: refactor into modules (#17266)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/lte_lc.rst

To enable the library, set the Kconfig option :kconfig:option:`CONFIG_LTE_LINK_CONTROL` to ``y`` in the project configuration file :file:`prj.conf`.

.. note::

>    This will enable the core features of the library related to network connectivity. To enable the other features, there are other Kconfig options that are explained later in this chapter.

Reviewer: ```suggestion
   This enables the core features of the library related to network connectivity.
   ```

Author: Is this enough? Any other way to stress that enabling the library enables only a limited featureset?

Reviewer: How about like this: By default, the library enables only the core features related to the network connectivity.

Author: Better, thanks.

   }

The code block demonstrates how you can use the library to asynchronously set up an LTE connection.

Additionally, to enable specific functionalities and receive specific events from the library, you must enable the corresponding modules through their respective Kconfig options:

Connection Parameters Evaluation:

>   Enable the :kconfig:option:`CONFIG_LTE_LC_CONN_EVAL` Kconfig option to enable all the following functionalities related to Connection Parameters Evaluation:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_CONN_EVAL` Kconfig option to enable all the following functionalities related to Connection Parameters Evaluation:
```

  * :c:func:`lte_lc_conn_eval_params_get`

eDRX (Extended Discontinuous Reception):

>   Enable the :kconfig:option:`CONFIG_LTE_LC_EDRX` Kconfig option to enable all the following functionalities related to eDRX:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_EDRX` Kconfig option to enable all the following functionalities related to eDRX:
```

  * :c:enumerator:`LTE_LC_EVT_EDRX_UPDATE` events
  * :c:func:`lte_lc_ptw_set`
  * :c:func:`lte_lc_edrx_param_set`
  * :c:func:`lte_lc_edrx_req`

>   * :c:func:`lte_lc_edrx_get`

Author: Agree. On a side note, we definitely need to bring some consistency in the Kconfig names in the future.

>   * :c:func:`lte_lc_edrx_get`

Reviewer 2: I'd probably list in the documentation all Kconfig options for a given module because of the inconsistent naming that we have to make things clear.

I haven't renamed the Kconfig's to avoid a breaking change because we do them too much already within NCS. It has been in my mind many times though.

Neighboring Cell Measurements:

>   Enable the :kconfig:option:`CONFIG_LTE_LC_NEIGHBOR_CELL_MEAS` Kconfig option to enable all the following functionalities related to Neighboring Cell Measurements:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_NEIGHBOR_CELL_MEAS` Kconfig option to enable all the following functionalities related to Neighboring Cell Measurements:
```

  * :c:enumerator:`LTE_LC_EVT_NEIGHBOR_CELL_MEAS` events
  * :c:func:`lte_lc_neighbor_cell_measurement_cancel`
  * :c:func:`lte_lc_neighbor_cell_measurement`

Periodic Search Configuration:

>   Enable the :kconfig:option:`CONFIG_LTE_LC_PERIODIC_SEARCH` Kconfig option to enable all the following functionalities related to Periodic Search Configuration:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_PERIODIC_SEARCH` Kconfig option to enable all the following functionalities related to Periodic Search Configuration:
```

  * :c:func:`lte_lc_periodic_search_request`
  * :c:func:`lte_lc_periodic_search_clear`
  * :c:func:`lte_lc_periodic_search_get`
  * :c:func:`lte_lc_periodic_search_set`

PSM (Power Saving Mode):

>   Enable the :kconfig:option:`CONFIG_LTE_LC_PSM` Kconfig option to enable all the following functionalities related to PSM:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_PSM` Kconfig option to enable all the following functionalities related to PSM:
```

  * :c:enumerator:`LTE_LC_EVT_PSM_UPDATE` events
  * :c:func:`lte_lc_psm_param_set`
  * :c:func:`lte_lc_psm_param_set_seconds`
  * :c:func:`lte_lc_psm_req`
  * :c:func:`lte_lc_psm_get`
  * :c:func:`lte_lc_proprietary_psm_req`

Release Assistance Indication (RAI):

>   Enable the :kconfig:option:`CONFIG_LTE_LC_RAI` Kconfig option to enable all the following functionalities related to RAI:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_RAI` Kconfig option to enable the following functionality related to RAI:
```

  * :c:enumerator:`LTE_LC_EVT_RAI_UPDATE` events

Modem Sleep Notifications:

>   Enable the :kconfig:option:`CONFIG_LTE_LC_MODEM_SLEEP` Kconfig option to enable all the following functionalities related to Modem Sleep Notifications:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_MODEM_SLEEP` Kconfig option to enable all the following functionalities related to Modem Sleep Notifications:
```

  * :c:enumerator:`LTE_LC_EVT_MODEM_SLEEP_EXIT_PRE_WARNING` events
  * :c:enumerator:`LTE_LC_EVT_MODEM_SLEEP_ENTER` events
  * :c:enumerator:`LTE_LC_EVT_MODEM_SLEEP_EXIT` events

Tracking Area Update (TAU) Pre-warning:

>   Enable the :kconfig:option:`CONFIG_LTE_LC_TAU_PRE_WARNING` Kconfig option to enable all the following functionalities related to TAU Pre-warning:

Reviewer: ```suggestion
  Use the :kconfig:option:`CONFIG_LTE_LC_TAU_PRE_WARNING` Kconfig option to enable the following functionality related to TAU Pre-warning:
```

   }

The code block demonstrates how you can use the library to asynchronously set up an LTE connection.

Additionally, to enable specific functionalities and receive specific events from the library, you must enable the corresponding modules through their respective Kconfig options:

Connection Parameters Evaluation:
  Use the :kconfig:option:`CONFIG_LTE_LC_CONN_EVAL` Kconfig option to enable the following functionality related to Connection Parameters Evaluation:

  * :c:func:`lte_lc_conn_eval_params_get`

eDRX (Extended Discontinuous Reception):
  Use the :kconfig:option:`CONFIG_LTE_LC_EDRX` Kconfig option to enable all the following functionalities related to eDRX:

  * :c:enumerator:`LTE_LC_EVT_EDRX_UPDATE` events
  * :c:func:`lte_lc_ptw_set`
  * :c:func:`lte_lc_edrx_param_set`
  * :c:func:`lte_lc_edrx_req`
  * :c:func:`lte_lc_edrx_get`

>   * :kconfig:option:`LTE_EDRX_REQ`

Reviewer 2: Needed to get hyperlink to work:
```suggestion
  * :kconfig:option:`CONFIG_LTE_EDRX_REQ`
```

Fix all kconfigs in this chapter.

Reviewer: Very good catch. I totally overlooked these.

Author: Thanks!

Enabling power-saving features
==============================

> To enable power-saving features, enable the following options:

Reviewer: ```suggestion
To enable power-saving features, use the following options:
```

Modem firmware version 1.3.0 and higher supports connection a pre-evaluation feature that allows the application to get information about a cell that is likely to be used for an RRC connection.
Based on the parameters received in the function call, the application can decide whether to send application data or not.

> To enable this module, enable the :kconfig:option:`CONFIG_LTE_LC_CONN_EVAL`.

Reviewer: ```suggestion
To enable this module, use the :kconfig:option:`CONFIG_LTE_LC_CONN_EVAL` Kconfig option.
```

> To enable modem sleep and TAU pre-warning notifications, enable the following options:

Reviewer: ```suggestion
To enable modem sleep and TAU pre-warning notifications, use the following options:
```
